src/uct/ib/ud/base/ud_iface.c

```diff
@@ -439,6 +439,7 @@ UCS_CLASS_INIT_FUNC(uct_ud_iface_t, uct_ud_iface_ops_t *ops, uct_md_h md,
     self->config.peer_timeout    = ucs_time_from_sec(config->peer_timeout);
     self->config.check_grh_dgid  = config->dgid_check &&
                                    uct_ib_iface_is_roce(&self->super);
+    self->config.wmax            = config->wmax;
```

Comment:
check that the value is valid (in range min..max)
of not use ucs_diag() to print error